### DISCUSSION TITLE
Variable compression methods for new files/data or specific files

### DISCUSSION THREAD

**Discussion by yfdyh000** [juicedata/juicefs#4027] [Ideas]
I want the compression method to be changeable after formatting the FS, because I might want to store compressible and uncompressible data in the same volume and save some CPU time.

I understand that this may be a distant and even difficult to achieve, this is just an idea to share.

If possible, I would expect it to be controlled by `config` command for new data, matching for ext name or filepath pattern, and even for file headers, some kind of attribute flag.

**Comment by davies** [comment]
That could be too complicated to implement. LZ4/ZSTD is fast enough that even the data is not compressible, so we can just use them.
